### PR TITLE
Don't attempt payments before on chain syncing

### LINKED ISSUE #1055: Don't attempt payments before on chain syncing
I knew this would backfire. Don't allow for payments to happen before syncing occurs. Because it locks up HTLC's with a bad block height.

https://github.com/MutinyWallet/mutiny-web/issues/913

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #1055

Adds an `AtomicBool` that we have synced this session. Not using the one in storage because we should check every new startup. 

Also noticed that we did not have all of our initial send logic for keysends so copied it over.

### CODE REVIEW

File: mutiny-core/src/nodemanager.rs [hunk 1]
```diff
@@ -1238,6 +1241,9 @@ impl<S: MutinyStorage> NodeManager<S> {
             return Err(e);
         }
 
+        // set has synced to true
+        self.has_done_initial_ldk_sync.swap(true, Ordering::SeqCst);
+
```
Comment (TonyGiorgio) [inline mutiny-core/src/nodemanager.rs:1246]: Payments before syncing LDK isn't the problem, the problem is BDK chain tip is not up to date. This needs to be moved down.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
Also needs to be reproduced and tested.

File: mutiny-core/src/nodemanager.rs (hunk 1 above)
Comment (benthecarman) [reply mutiny-core/src/nodemanager.rs:1246]: I'm confused, how would bdk effect ldk? They don't talk to each other

Comment (TonyGiorgio) [reply mutiny-core/src/nodemanager.rs:1246]: how is the chain tip retrieved?

Comment (benthecarman) [reply mutiny-core/src/nodemanager.rs:1246]: ldk syncs it

**Review by TonyGiorgio — APPROVED** [review]
I'm trusting that you reproduced and tested this first before merging.

**Pull request merged**